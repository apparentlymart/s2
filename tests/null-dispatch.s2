#-*-c++-*-

layerinfo "type" = "core";
layerinfo "majorversion" = 0;

class Tree {  function grow; }
class Maple extends Tree { }
class Oak extends Tree {  }

function Tree::grow { "Tree grows.\n"; }
function Maple::grow { "Maple grows.\n"; }

function main ()
{
  var Tree t;
  $t = new Maple;  $t->grow();
  $t = new Oak;    $t->grow();
  var Tree notree; 
  $notree->grow();
  $notree->grow();
  $notree->grow();
}


