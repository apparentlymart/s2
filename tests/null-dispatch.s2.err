Method called on null Tree object at layer #1, line 19.