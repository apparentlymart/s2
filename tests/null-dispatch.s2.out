Maple grows.
Tree grows.
ERROR: Died in S2::run_code running main(): Method called on null Tree object at layer #1, line 19.